hangups/http_utils.py

```diff
@@ -5,57 +5,126 @@
 import collections
 import logging
 
-from hangups import exceptions
+from hangups import exceptions, channel
 
 logger = logging.getLogger(__name__)
 CONNECT_TIMEOUT = 30
 REQUEST_TIMEOUT = 30
 MAX_RETRIES = 3
 
-FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body',
-                                                         'cookies'])
+FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body'])
 
 
-@asyncio.coroutine
-def fetch(method, url, params=None, headers=None, cookies=None, data=None,
-          connector=None):
-    """Make an HTTP request.
+class ClientSession(aiohttp.ClientSession):
```

Comment:
What would be the advantage?

Additional ``close`` and ``request`` methods, plus extended access-paths in the current methods are needed.